File: NTTU_AlgoHomework/HW1-1.py
# HWP1-1
# Author / KevinLee, TrevorWu
# Date / 20180920

# Clean Treminal Screen
# Now can auto detect Windows or macOS
# import os, platform
# if platform.system() == 'Windows':
#     os.system('cls')
# else:
#     os.system('clear')

# number of test cases
number = int(input())

# Loop by number of test cases
for a in range(number):
# while number != 0:
# 	number -= 1
	# Claen how many times to swaps.
	times = 0

	# Useless
	input()

	# Enter test cases and split it by space
	arr = input().split()

	# Write each number into array
	arr = [int(item) for item in arr]

	# Start Compar
	for i in range(len(arr)):
			# echo every elements in list
			for j in range(len(arr)-i-1):
		        # Change Order
				if arr[j] > arr[j+1]:
					arr[j], arr[j+1] = arr[j+1], arr[j]
					times += 1
	print ('Optimal train swapping takes', times, 'swaps.')

File: NTTU_AlgoHomework/HW1-1.cpp
#include<iostream>
#include<vector>
using namespace std;
int main(){
	int ts;cin>>ts;
	while(ts--){
		int n,ans=0;cin>>n;
		vector<int> a(n);
		for(int k=0;k<n;k++)cin>>a[k];
		for(int k=0;k<n;k++){
			for(int h=0;h<k;h++){
				if(a[h]>a[k])ans++;
			}
		}
		cout<<"Optimal train swapping takes "<<ans<<" swaps."<<endl;
	}
} 